src/evm-block-extractor/src/storage_clients/gcp_big_query.rs

```diff
@@ -0,0 +1,260 @@
+use ethers_core::types::{Block, Transaction, TransactionReceipt, H256};
+use gcp_bigquery_client::model::query_request::QueryRequest;
+use gcp_bigquery_client::model::table_data_insert_all_request::TableDataInsertAllRequest;
+use gcp_bigquery_client::model::table_data_insert_all_request_rows::TableDataInsertAllRequestRows;
+use gcp_bigquery_client::Client;
+use serde::Serialize;
+
+use super::BlockChainDB;
+use crate::constants::{BLOCKS_TABLE_ID, PROJECT_ID, RECEIPTS_TABLE_ID};
+
+#[derive(Clone)]
+/// A client for BigQuery that can be used to query and insert data
+pub struct BigQueryBlockChain {
+    client: Client,
+    /// The project ID of the BigQuery project
+    project_id: String,
+    /// The dataset ID of the BigQuery table
+    ///
+    /// Can be mainnet/testnet
+    dataset_id: String,
+}
+
+impl BigQueryBlockChain {
+    /// Creates a new BigQuery client
+    /// The service account key should be stored in the
+    /// `GCP_BLOCK_EXTRACTOR_SA_KEY` environment variable
+    pub async fn new(dataset_id: String, sa_key: String) -> anyhow::Result<Self> {
+        let service_account = yup_oauth2::parse_service_account_key(sa_key)?;
+
+        let client = Client::from_service_account_key(service_account, false).await?;
+
+        Ok(Self {
+            client,
+            project_id: PROJECT_ID.to_string(),
+            dataset_id: dataset_id.to_string(),
+        })
+    }
+
+    async fn execute_query<T: for<'de> serde::Deserialize<'de>>(
```

Comment:
How did you test this one? Are you aware of BigQueryEmulator? (`docker pull ghcr.io/goccy/bigquery-emulator:latest`)